### PR TITLE
[SPARK-31778][K8S][BUILD] Support cross-building docker images

### PR DESCRIPTION
**Pull request by holdenk**
### What changes were proposed in this pull request?

Add cross build support to our docker image script using the new dockerx extension.

### Why are the changes needed?

We have a CI for Spark on ARM, we should support building images for ARM and AMD64.

### Does this PR introduce _any_ user-facing change?

Yes, a new flag is added to the docker image build script to cross-build

### How was this patch tested?
Manually ran build script & pushed to https://hub.docker.com/repository/registry-1.docker.io/holdenk/spark/tags?page=1 verified amd64 & arm64 listed.

### CODE REVIEW

File: bin/docker-image-tool.sh [hunk 1]
```diff
@@ -212,23 +230,24 @@ Builds or pushes the built-in Spark Docker image.
...
   build       Build image. Requires a repository address to be provided if the image will be
-              pushed to a different registry.
+	      pushed to a different registry.
   push        Push a pre-built image to a registry. Requires a repository address to be provided.
 
 Options:
   -f file               Dockerfile to build for JVM based Jobs. By default builds the Dockerfile shipped with Spark.
   -p file               (Optional) Dockerfile to build for PySpark Jobs. Builds Python dependencies and ships with Spark.
-                        Skips building PySpark docker image if not specified.
+			Skips building PySpark docker image if not specified.
```
Comment (dongjoon-hyun) [inline bin/docker-image-tool.sh:239]: Please remove `TAB` here~ :)

Comment (holdenk) [reply bin/docker-image-tool.sh]: Good catch, my emacs shell mode was overriding my untabify

**Review by dongjoon-hyun — DISMISSED** [review]
This doesn't work on `Docker Desktop`. I tested on the latest one, `Docker Desktop Edge 2020-05-20 2.3.1.0`. In this case, we had better describe the limitation of this new feature. Could you describe the used and expected environment in your PR description?
```
$ bin/docker-image-tool.sh -r docker.io/myrepo -t v3.0.0 -X -b java_image_tag=11-jre-slim build
...
+ docker buildx build --platform linux/amd64,linux/arm64 --build-arg java_image_tag=11-jre-slim -t docker.io/myrepo/spark:v3.0.0 -f kubernetes/dockerfiles/spark/Dockerfile .
multiple platforms feature is currently not supported for docker driver. Please switch to a different driver (eg. "docker buildx create --use")
```

**Comment by holdenk** [comment]
It does work on docker desktop but you need to be setup to cross build (currently experimental). I'll add a link in the script for those instructions.

**Comment by dongjoon-hyun** [comment]
Given the new URL at the last commit, it seems that this feature is pretty immature. Can we wait until it become stable?

BTW, I've also been in `experimental` mode as described in the doc, but the situation is the same. If this is not working at out-of-box status, the single URL looks insufficient to support this PR.
```
$ cat ~/.docker/config.json | grep experimental
  "experimental" : "enabled",
```

**Comment by holdenk** [comment]
Cross building is relatively new, but a lot of docker images are cross built already. It’s optional and disabled by default (eg to have this build you’d have to explicitly configure it).

For example the JDK11 base image is already cross built on amd64 and aarm64.

You need to follow the buildx setup instructions not just turn on the expiremental channel.

**Comment by dongjoon-hyun** [comment]
Okay. I removed my previous change request.
I'll leave this to the other reviewers then. Thanks.

**Pull request closed without merging**